Jenkinsfile

```diff
@@ -1,15 +1,28 @@
 pipeline {
-
-    agent { label 'linux-slave' }
+    agent none
 
     stages {
-        stage('Run Tests') {
+        stage('Input') {
+            when {
+                branch 'master'
+                beforeInput true
+            }
+            steps {
+                script {
+                    env.DEPLOY_CLIENT = input \
+                        message: 'Would you like to deploy the client to NPM?', \
+                        parameters: [choice(name: 'DEPLOY_CLIENT', choices: ['Yes', 'No'], description: '')]
```

Comment:
how does this work with the new githook stuff? Would we need to go into Jenkins each time and click a button? Could we have some kind of timeout to a default response?

Reply:
The build would start as soon as the change is pushed, but will then pause and wait for input. The build will remain paused until someone goes into the Jenkins UI and submits this form. This will _only_ happen when merging to `master` though - other builds will run without manual action required, as happens currently.

Since they're only when we're modifying production, I don't think we need any timeout on these kind of actions. If anything, requiring manual approval on these builds is a bonus.

Reply:
Ah ok. If it's only going to prompt for master then that's fine. It's unlikely we'd be doing an unattended production release